### PR TITLE
yarpcerrors: Tweak godocs

### PR DESCRIPTION
**Pull request by abhinav**
This tweaks some of the docs in yarpcerrors. Rationale behind the
changes is to reduce docs that,

- are just the code in prose form
- lay out too many implementation details
- mention future functionality or TODOs

### CODE REVIEW

**Comment by abhinav** [comment]
@peter-edge or @willhug, unless you have concerns, I'm going to merge this today.

**Comment by peter-edge** [comment]
Sorry - go for it.

File: yarpcerrors/errors.go [hunk 1]
```diff
@@ -57,7 +55,9 @@ func FromError(err error) *Status {
 	}
 }
 
-// IsStatus returns true if err is a Status.
+// IsStatus returns whether the provided error is a YARPC error.
```
Comment (peter-edge) [inline yarpcerrors/errors.go]: Why is this changed? You seem to clarify in `FromError` that this relies on this being a `Status`, but then remove it here.

**Comment by abhinav** [comment]
Thanks

File: yarpcerrors/errors.go (hunk 1 above)
Comment (peter-edge) [reply yarpcerrors/errors.go]: Eh, this is fine, let's just leave it.
